wig/wig.go

```diff
@@ -1,115 +1,107 @@
+	}
+	return finalWig
+}
 
-			var lineFields []string = strings.Fields(line)
-			var chromList []string = strings.Split(lineFields[1], "=")
-			currentWig.Chrom = chromList[1]
+func NextWig(file *fileio.EasyReader) (Wig, bool) {
+	var line string
+	var currentWig Wig
+	var doneReading bool
+	var peek []byte
+	var err error
+	var lineFields, chromList, startList, stepList []string
 
-		} else if lineFields[0] == "fixedStep" {
+	for line, doneReading = fileio.EasyNextRealLine(file); !doneReading; line, doneReading = fileio.EasyNextRealLine(file) {
+		if strings.HasPrefix(line, "fixedStep") {
+			lineFields = strings.Fields(line)
 			if len(lineFields) != 4 {
 				log.Fatalf("Invalid number of arguments, expecting 4, received %d\n", len(lineFields))
 			}
-
-			currentWig = new(Wig)
-			answer = append(answer, currentWig)
-
-			var lineFields []string = strings.Fields(line)
-			var chromList []string = strings.Split(lineFields[1], "=")
+			currentWig.StepType = "fixedStep"
+			chromList = strings.Split(lineFields[1], "=")
 			currentWig.Chrom = chromList[1]
-			var startList []string = strings.Split(lineFields[2], "=")
+			startList = strings.Split(lineFields[2], "=")
 			currentWig.Start = common.StringToInt(startList[1])
-			var stepList []string = strings.Split(lineFields[3], "=")
+			stepList = strings.Split(lineFields[3], "=")
 			currentWig.Step = common.StringToInt(stepList[1])
-			currentWig.StepType = "fixedStep"
 
+		} else if strings.HasPrefix(line, "variableStep") {
+			log.Fatalf("ERROR: %s is variableStep Wig, must convert to fixedStep before reading in Wig to gonomics", file.File.Name())
 		} else {
-			//check length = 2 inputs is variable, 1 is fixed
-			var lineFields []string = strings.Fields(line)
-			currentValue := new(WigValue)
-
-			if len(lineFields) == 1 {
-				//fixedStep
-				currentValue.Value = common.StringToFloat64(lineFields[0])
-				currentValue.Position = currentWig.Start + len(currentWig.Values)*currentWig.Step
-			} else if len(lineFields) == 2 {
-				//variableStep
-				currentValue.Position = common.StringToInt(lineFields[0])
-				currentValue.Value = common.StringToFloat64(lineFields[1])
+			if currentWig.StepType == "" {
+				log.Fatalf("ERROR: %s is missing a wig header (e.g. fixedStep chrom=chr...)", file.File.Name())
 			}
+			currentWig.Values = append(currentWig.Values, common.StringToFloat64(line))
+		}
 
-			currentWig.Values = append(currentWig.Values, currentValue)
+		peek, err = fileio.EasyPeekReal(file, 1)
+		if errors.Is(err, io.EOF) {
+			return currentWig, doneReading // we could return true here, but we do not in order to match the behavior of other Next functions.
+		} else {
+			exception.PanicOnErr(err)
+		}
+		if peek[0] == 'f' {
+			return currentWig, doneReading
 		}
 	}
-	return answer
+	return currentWig, doneReading
 }
 
 //Prints the first record in a Wig struct. Mainly used for debugging.
-func PrintFirst(rec []*Wig) {
+func PrintFirst(rec []Wig) {
```

Comment:
Maybe this can disappear?  Or get moved to a debug.go file?  Not for now, but just leaving my thoughts.

Reply:
Interested in talking with you more about a "debug.go" file at a future meeting.